### PR TITLE
Initial project setup

### PR DESCRIPTION
**Pull request by michal-lepik**

### CODE REVIEW

File: frontend/dev.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM node:dubnium
+
+WORKDIR /frontend
+
+COPY . .
```
Comment (grzegorz-bielski) [inline frontend/dev.Dockerfile]: First copy package.json+lock, then install all deps, then copy rest of the code. Thanks to that all deps will be cached in the previous temp container layer and you get faster builds.

File: nginx/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM nginx
+
+WORKDIR /nginx
+
+RUN rm /etc/nginx/conf.d/*
+COPY nginx.conf /etc/nginx/conf.d/nginx.conf
```
Comment (grzegorz-bielski) [inline nginx/Dockerfile:6]: Use multi-stage builds for the minimal final image.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      context: ./nginx
+    ports:
+      - 4200:80
+    restart: always
+    depends_on:
+      - frontend
+    networks:
+      - frontend
+
+networks:
```
Comment (grzegorz-bielski) [inline docker-compose.yml]: is that necessary? You can remove it and live within a default network

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,5 @@
+build-dev:
+	docker-compose build
```
Comment (grzegorz-bielski) [inline Makefile]: - Use `--parallel` for faster builds. 
- Add a command for installing node-deps locally. 
```
install-local-dependencies:
	npm --prefix ./frontend/ install
```

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+
+dev:
+	docker-compose up
```
Comment (grzegorz-bielski) [inline Makefile]: add ` --remove-orphans`

Comment (grzegorz-bielski) [reply Makefile]: Also, where is `docker-compose` for prod/staging? Please create it in this PR.

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+services:
+  frontend:
+    image: instapound_frontend
+    container_name: instapound-frontend
+    build:
+      context: ./frontend
+      dockerfile: dev.Dockerfile
+    volumes:
+      - ./frontend:/frontend
```
Comment (grzegorz-bielski) [inline docker-compose.yml]: Use `:delegated` for better performance in the dev volumes: `./frontend:/frontend:delegated`

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    networks:
+      - frontend
+
+  nginx:
+    image: instapound_nginx
+    container_name: instapound-nginx
+    build:
+      context: ./nginx
+    ports:
+      - 4200:80
```
Comment (grzegorz-bielski) [inline docker-compose.yml:16]: You might consider moving this to the `$HOSTPORT` env ... or not (I'd do that).

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    restart: always
```
Comment (grzegorz-bielski) [inline docker-compose.yml]: This can be annoying, you might want to use `unless-stopped`

**Comment by grzegorz-bielski** [comment]
In addition, create `.dockerignore` that have at least this stuff (for JS Project):
```
node_modules
npm-debug.log
Dockerfile*
.dockerignore
.git
README
LICENSE
```
Thank to that the docker won't load `node_modules` when creating images (during `COPY`), so the builds will be faster.

**Review by grzegorz-bielski — CHANGES_REQUESTED** [review]
/\ comments

File: .dockerignore [hunk 9]
```diff
@@ -0,0 +1,7 @@
+node_modules
+npm-debug.log
+Dockerfile*
```
Comment (mateuszkozakiewicz) [inline .dockerignore]: Your files are named for example dev.Dockerfile so here should be *Dockerfile

**Review by mateuszkozakiewicz — APPROVED** [review]

**Pull request merged**